docs/user-guide/configuring.md

```diff
@@ -1070,12 +1070,26 @@ Of particular note is that like `.gitignore` files, all paths used as patterns f
 
 Please see `.gitignore`'s specification for further examples of valid syntax.
 
-In addition to any patterns in a `.eslintignore` file, ESLint ignores files in `/**/node_modules/*` by default. It can still be added using `!`.
+In addition to any patterns in the `.eslintignore` file, ESLint always follows a couple implicit ignore rules even if the `--no-ignore` flag is passed. The implicit rules are as follows:
 
-For example, placing the following `.eslintignore` file in the current working directory will not ignore `node_modules/*` and ignore anything in the `build/` directory except `build/index.js`:
+* `node_modules/` as well as `bower_components/` directories are ignored.
+* Dotfiles as well as Dotfolders and their contents are ignored.
+
+There are also some exceptions to these rules:
+
+* If the path to lint is a glob pattern or directory path and contains a Dotfolder, all Dotfiles and Dotfolders will be linted. This includes subDotfiles and subDotfolders that are buried deeper in the directory structure.
+
+  For example, `eslint .config/` will lint all Dotfolders and Dotfiles in the `.config` directory, including immediate children as well as children that are deeper in the directory structure.
+
+* If the path to lint is a specific file path and the `--no-ignore` flag has been passed, ESLint will lint the file regardless of the implicit ignore rules.
+
+  For example, `eslint .config/my-config-file.js --no-ignore` will cause `my-config-file.js` to be linted. It should be noted that the same command without the `--no-ignore` line will not lint the `my-config-file.js` file.
+
+Another example: placing the following `.eslintignore` file in the current working directory will ignore the `.git`, `node_modules`, and `bower_components` directories in the project root and anything in the `build/` directory, except `build/index.js`:
 
 ```text
-# node_modules/* is ignored by default, but can be added using !
+# All .git/* Dotfiles as well as /node_modules/* and /bower_components/* in the project root are ignored by default
```

Comment:
Technically, all dotfiles and dotfolders (except for .eslintrc.*) are ignored, not just `.git/*.